flake.nix

```diff
@@ -118,12 +124,16 @@
           cargo-insta
           cargo-nextest
           cargo-watch
-        ];
+
+          # Extra, more specific tools follow
+        ] ++ pkgs.lib.optionals useMoldLinker [ mold ];
 
         shellHook = ''
           export RUST_BACKTRACE=1
           export ZSTD_SYS_USE_PKG_CONFIG=1
           export LIBSSH2_SYS_USE_PKG_CONFIG=1
+        '' + pkgs.lib.optionalString useMoldLinker ''
+          export RUSTFLAGS="-C link-arg=-fuse-ld=mold"
```

Comment:
Should this include the previous `$RUSTFLAGS`?

Reply:
It's possible for you to want to transitively apply flags, I think; so yes, good catch. Fixed